addons/binding/org.openhab.binding.neato/src/main/java/org/openhab/binding/neato/handler/NeatoHandler.java

```diff
@@ -0,0 +1,257 @@
+            logger.debug("Ok - will handle command for CHANNEL_COMMAND");
+
+            try {
+                mrRobot.sendCommand(command);
+            } catch (Exception exc) {
+                logger.error("Error while processing command from openHAB. Error: {}", exc.getMessage());
+            }
+
+        }
+        this.refreshStateAndUpdate();
+    }
+
+    @Override
+    public void initialize() {
+
+        logger.info("Will boot up Neato Vacuum Cleaner binding!");
+
+        updateStatus(ThingStatus.OFFLINE);
+        thisStatus = ThingStatus.OFFLINE;
+
+        Configuration config = getThing().getConfiguration();
+        vacuumSecret = (String) config.get(CONFIG_SECRET);
+        vacuumName = (String) config.get(CONFIG_NAME);
+        vacuumSerialNumber = (String) config.get(CONFIG_SERIAL);
+
+        logger.info("Got config settings for {} with serial number: {}", vacuumName, vacuumSecret);
+
+        super.initialize();
+
+        try {
+            refreshTime = ((BigDecimal) config.get(CONFIG_REFRESHTIME)).intValue();
+            if (refreshTime <= 0) {
+                throw new IllegalArgumentException("Refresh time must be positive number!");
+            }
+        } catch (Exception e) {
+            logger.warn("Refresh time [{}] is not valid. Falling back to default value: {}. {}", refreshTime,
+                    DEFAULTREFRESHTIME, e.getMessage());
+            refreshTime = DEFAULTREFRESHTIME;
+        }
+
+        mrRobot = new NeatoRobot(vacuumSerialNumber, vacuumSecret, vacuumName);
+        try {
+
+            this.startAutomaticRefresh();
+
+        } catch (Exception ex) {
+            // Note: When initialization can NOT be done set the status with more details for further
+            // analysis. See also class ThingStatusDetail for all available status details.
+            // Add a description to give user information to understand why thing does not work
+            // as expected. E.g.
+            // updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+            // "Can not access device as username and/or password are invalid");
+
+            logger.error(ex.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, ex.getMessage());
+            thisStatus = ThingStatus.OFFLINE;
+
+        }
+
+    }
+
+    public void refreshStateAndUpdate() {
+        try {
+
+            if (mrRobot.sendGetState()) {
+
+                if (thisStatus != ThingStatus.ONLINE) {
+                    updateStatus(ThingStatus.ONLINE);
+                    thisStatus = ThingStatus.ONLINE;
+                }
+
+                mrRobot.sendGetGeneralInfo();
+
+                List<Channel> channels = getThing().getChannels();
+
+                for (Channel channel : channels) {
+                    publishChannelIfLinked(channel.getUID());
+                }
+
+            }
+        } catch (Exception e) {
```

Comment:
Do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception